asv/util.py

```diff
@@ -272,20 +272,18 @@ def human_value(value, unit, err=None):
     return display
 
 
-def which(filename):
+def which(filename, paths=None):
     """
     Emulates the UNIX `which` command in Python.
 
     Raises an IOError if no result is found.
     """
-    if WIN:
-        if not filename.endswith('.exe'):
-            filename = filename + '.exe'
-
     if os.path.sep in filename:
         locations = ['']
     else:
         locations = os.environ.get("PATH", "").split(os.pathsep)
+        if paths is not None:
+            locations = paths + locations
 
         if WIN:
             # On windows, an entry in %PATH% may be quoted
```

Comment:
Also, this part can be run only when `paths is not None` because the `paths` list does not contain quoted paths.